File: source/java/nl/esciencecenter/xenon/examples/schedulers/SlurmQueuesGetter.java
.. code-tab:: java

    import nl.esciencecenter.xenon.credentials.PasswordCredential;
    import nl.esciencecenter.xenon.schedulers.Scheduler;

    public class SlurmQueuesGetter {

        public static void main(String[] args) throws Exception {

            String adaptor = "slurm";
            String location = "localhost:10022";

            // make the password credential for user 'xenon'
            String username = "xenon";
            char[] password = "javagat".toCharArray();
            PasswordCredential credential = new PasswordCredential(username, password);

            // create the SLURM scheduler
            Scheduler scheduler = Scheduler.create(adaptor, location, credential);

            // ask SLURM for its queues
            System.out.println("Available queues (starred queue is default):");
            for (String queueName : scheduler.getQueueNames()) {
                if (queueName == scheduler.getDefaultQueueName()) {
                    System.out.println(queueName + "*");
                } else {
                    System.out.println(queueName);
                }
            }
        }
    }

File: source/python/examples/schedulers/slurm_queues_getter.py
.. code-tab:: python

    from xenon import Server, PasswordCredential

    with Server() as xenon:
        location = 'localhost:10022'
        credential = PasswordCredential(
                username='xenon',
                password='javagat')

        scheduler = xenon.create_scheduler(
                adaptor='slurm',
                location=location,
                credential=credential)

        print("Available queues (starred queue is default):")
        default_queue = scheduler.get_default_queue_name()
        for queue_name in scheduler.get_queues():
            if queue_name == default_queue:
                print("{}*".format(queue_name))
            else:
                print(queue_name)

        scheduler.close()
